WooFoundation/WooFoundation/Extensions/UIImage+Background.swift

```diff
@@ -0,0 +1,24 @@
+import UIKit
+
+public extension UIImage {
+    /// Adds a background color to the given UIImage, setting also whether it should be opaque or not
+    /// 
+    func withBackground(color: UIColor, opaque: Bool = true) -> UIImage {
+      UIGraphicsBeginImageContextWithOptions(size, opaque, scale)
+
+      guard let ctx = UIGraphicsGetCurrentContext(),
+                let image = cgImage else {
+          return self
+      }
+
+      defer { UIGraphicsEndImageContext() }
+
+      let rect = CGRect(origin: .zero, size: size)
+      ctx.setFillColor(color.cgColor)
+      ctx.fill(rect)
+      ctx.concatenate(CGAffineTransform(a: 1, b: 0, c: 0, d: -1, tx: 0, ty: size.height))
+      ctx.draw(image, in: rect)
+
+      return UIGraphicsGetImageFromCurrentImageContext() ?? self
+    }
```

Comment:
There's a lot happening here, so I think we could improve it by:
-  Renaming `ctx` to `currentGraphicsContext`
- Adding internal comments
- Changing:
```diff
-  ctx.concatenate(CGAffineTransform(a: 1, b: 0, c: 0, d: -1, tx: 0, ty: size.height))
+ // Flips the context vertically, then translates it vertically
+ // By default, the coordinate system in Core Graphics is different from that of UIKit
+ currentGraphicsContext.scaleBy(x: 1, y: -1)
+ currentGraphicsContext.translateBy(x: 0, y: -size.height)
```
```suggestion
     func withBackground(color: UIColor, opaque: Bool = true) -> UIImage {
        UIGraphicsBeginImageContextWithOptions(size, opaque, scale)

        // Checks if there is a current graphics context, and the UIImage has a CGImage,
        // return the original image otherwise
        guard let currentGraphicsContext = UIGraphicsGetCurrentContext(),
              let image = cgImage else {
            return self
        }

        // Ensure removal of custom currentGraphicsContext on deallocation
        defer { UIGraphicsEndImageContext() }

        let rect = CGRect(origin: .zero, size: size)
        currentGraphicsContext.setFillColor(color.cgColor)
        currentGraphicsContext.fill(rect)
        // Flips the context vertically, then translates it vertically
        // By default, the coordinate system in Core Graphics is different from that of UIKit
        currentGraphicsContext.scaleBy(x: 1, y: -1)
        currentGraphicsContext.translateBy(x: 0, y: -size.height)
        currentGraphicsContext.draw(image, in: rect)

        // Return the UIImage from the current context,
        // of the original image if happens to fail.
        return UIGraphicsGetImageFromCurrentImageContext() ?? self
    }
```

What do you think? I tested the changes to `CGAfflineTransform` and we get the same result , unless there's any other concern I've missed it seems that, from Apple docs, we shouldn't access this property directly.

Reply:
Thanks for the comment! I added some of the suggestions here in 0d17ba5857ffb62292990fb5c0801de4ad120cda Some parts of the code were self-explanatory and didn't need comments, but some others were indeed abstruse